### ISSUE TITLE
Ableton doesn't "do" anything when I send it the following notes.

### ISSUE THREAD

**Issue by willemodendaal** [haubie/midiex#1]
Such an exciting project! Very keen to use Elixir to do some midi things.

I'm sure it's me doing something wrong, not Ableton, just not sure what... (First time generating midi with code)

When I run the following code, I can see that Ableton is receiving something (top right flashing icon while this code runs).

But I can't get Ableton to play (monitor) or record midi notes. I think I've done the necessary configuration, including selecting the appropriate midi "device" that has the expected name I gave to it. The midi track plays fine if I test with the computer keyboard.

Is there something obviously wrong with the code below? Intended to play middle-C for as long as the program runs (toggling every 250ms):

```
  def testing_testing_123(device_name, nr_seconds \\ 5) do
    IO.puts("Starting test, will run for #{nr_seconds} seconds...")
    # List MIDI ports
    Midiex.ports()

    # Create a virtual output connection
    piano = Midiex.create_virtual_output(device_name)

    # Send to MIDI messages to a connection (Middle C, at full velocity)
    # On channel "1" (bc. "91" and "81")
    note_on = <<0x91, 60, 127>>
    note_off = <<0x81, 60, 127>>

    # Press and release C, twice a second.
    for _n <- 1..(nr_seconds * 2) do
      IO.puts("Middle C....")
      Midiex.send_msg(piano, note_on)
      :timer.sleep(250)
      Midiex.send_msg(piano, note_off)

      IO.puts("Middle C... again ...")
      Midiex.send_msg(piano, note_on)
      :timer.sleep(250)
      Midiex.send_msg(piano, note_off)
    end

    IO.puts("Show's over!")
  end

```

Not expecting support for Ableton, but maybe this gives a clue? Anything amiss here?

<img width="106" alt="image" src="https://github.com/haubie/midiex/assets/915354/dc4c4098-55bf-4e94-86d1-1184c2efa76d">

**Comment by willemodendaal** [comment]
I checked again, after 24 hours, and it works!

So a restart somewhere must have fixed it.

Thanks for this library, love it, time to make some tunes.

**Comment by haubie** [comment]
Thanks @willemodendaal, I haven't yet tried it with Ableton and very useful reading about your experience.

**Comment by thbar** [comment]
For some reason, I faced the same symptoms, and I'm not sure what fixed it either at this point.

My code is a bit different : I removed virtual output because I re-start the script constantly, and instead used IAC:

```elixir
# https://hexdocs.pm/midiex/readme.html
port = Midiex.ports(~r/IAC Driver/, :output)
piano = Midiex.open(port)
note_on = <<0x90, 57, 127>>
# note_on = <<0x90, 57+6, 127>>
_note_off = <<0x80, 57, 127>>

Midiex.send_msg(piano, note_on)
```

Will report back if I figure this out!

**Comment by haubie** [comment]
Thanks @thbar.

I've just tried to reproduce this issue with a fresh install of Ableton Lite 12.0.2 on Mac OS but I've been unsuccessful in repeating it.

I'm naive when it comes to the workings of Ableton but so far it seems to correctly pick up the MIDI data sent from Midiex.   I've tested it by sending directly to a MIDI port from the IAC driver (as you have) as well as by creating a virtual output (similar to @willemodendaal's example).

Let me know if it happens again!
